src/bmffimage.cpp

```diff
@@ -0,0 +1,330 @@
+#include <cstring>
+#include <iostream>
+#include <cassert>
+#include <cstdio>
+
+struct BmffBoxHeader
+{
+    uint32_t length;
+    uint32_t type;
+};
+
+#define ID(string) ((string[0] << 24) | (string[1] << 16) | (string[2] << 8) | string[3])
+
+#define TAG_ftyp ID("ftyp") /**< File type box */
+#define TAG_avif ID("avif") /**< AVIF */
+#define TAG_heic ID("heic") /**< HEIF */
+#define TAG_heif ID("heif") /**< HEIF */
+#define TAG_crx  ID("crx ") /**< Canon CR3 */
+#define TAG_moov ID("moov") /**< Movie */
+#define TAG_meta ID("meta") /**< Metadata */
+#define TAG_mdat ID("mdat") /**< Media data */
+#define TAG_uuid ID("uuid") /**< UUID */
+#define TAG_dinf ID("dinf") /**< Data information */
+#define TAG_iprp ID("iprp") /**< Item properties */
+#define TAG_ipco ID("ipco") /**< Item property container */
+#define TAG_iinf ID("iinf") /**< Item info */
+#define TAG_iloc ID("iloc") /**< Item location */
+#define TAG_ispe ID("ispe") /**< Image spatial extents */
+
+// *****************************************************************************
+// class member definitions
+namespace Exiv2
+{
+    static bool enabled = false;
+
+    EXIV2API bool enableBMFF(bool enable)
+    {
+#ifdef EXV_ENABLE_BMFF
+        enabled = enable;
+        return true;
+#endif // EXV_ENABLE_BMFF
+        enable = false; // unused
+        return enable;
+    }
+
+    BmffImage::BmffImage(BasicIo::AutoPtr io, bool /* create */)
+            : Image(ImageType::bmff, mdExif | mdIptc | mdXmp, io)
+    {
+    } // BmffImage::BmffImage
+
+    std::string BmffImage::toAscii(long n)
+    {
+        const char* p = (const char*) &n;
+        std::string result;
+        bool bBigEndian = isBigEndianPlatform();
+        for ( int i = 0 ; i < 4 ; i++) {
+            result += p[ bBigEndian ? i : (3-i) ];
+        }
+        return result;
+    }
+
+    std::string boxName(uint32_t box)
+    {
+        char           name[5];
+        std::memcpy   (name,&box,4);
+        name[4] = 0   ;
+        return std::string(name) ;
+    }
+
+    static void boxes_check(size_t b, size_t m)
+    {
+        if ( b > m ) {
+#ifdef EXIV2_DEBUG_MESSAGES
+            std::cout << "Exiv2::BmffImage::readMetadata box maximum exceeded" << std::endl;
+#endif
+            throw Error(kerCorruptedMetadata);
+        }
+    }
+
+    bool superBox(uint32_t box)
+    {
+        return      box == TAG_moov
+                ||  box == TAG_dinf
+                ||  box == TAG_iprp
+                ||  box == TAG_ipco
+                ||  box == TAG_meta
+                ||  box == TAG_iinf
+                ||  box == TAG_iloc
+        ;
+    }
+
+    bool fullBox(uint32_t box)
+    {
+        return      box == TAG_meta
+                ||  box == TAG_iinf
+                ||  box == TAG_iloc
+        ;
+    }
+
+    std::string BmffImage::mimeType() const
+    {
+        switch (fileType)
+        {
+            case TAG_avif:
+                return "image/avif";
+
+            case TAG_heic:
```

Comment:
Seems there is dedicated image/heic mime type: https://www.iana.org/assignments/media-types/media-types.xhtml#image